database/user_test.go

```diff
@@ -0,0 +1,42 @@
+package database
+
+import (
+	"testing"
+	"time"
+)
+
+// TestMonthStart ensures we calculate the start of the subscription month
+// correctly.
+func TestMonthStart(t *testing.T) {
+	now := time.Now()
+	tests := []struct {
+		subUntil     time.Time
+		startOfMonth time.Time
+	}{
+		{
+			// If the monthly sub cycle reset yesterday, expect the start of
+			// month to also be yesterday.
+			subUntil:     time.Date(2020, 1, now.Day()-1, 3, 4, 5, 6, time.UTC),
```

Comment:
f/u: Test with different months and years apart from just January and 2020

Reply:
Maybe worth adding an edge case for end of month, start of month, and leap years?

Reply:
GitHub is rather unhelpful with follow ups - I need to link it manually. F/U: #48